acceptance-tests/src/test/java/io/blueocean/ath/offline/BitbucketServerTest.java

```diff
@@ -0,0 +1,66 @@
+package io.blueocean.ath.offline;
+
+import com.cdancy.bitbucket.rest.BitbucketClient;
+import com.cdancy.bitbucket.rest.options.CreateRepository;
+import com.google.inject.Inject;
+import io.blueocean.ath.ATHJUnitRunner;
+import io.blueocean.ath.Login;
+import io.blueocean.ath.WaitUtil;
+import io.blueocean.ath.pages.blue.DashboardPage;
+import org.apache.log4j.Logger;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.openqa.selenium.By;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+import java.io.IOException;
+
+@Login
+@RunWith(ATHJUnitRunner.class)
+public class BitbucketServerTest {
+
+    private static Logger LOGGER = Logger.getLogger(BitbucketServerTest.class);
+
+    @Inject
+    WaitUtil wait;
+
+    @Inject
+    DashboardPage dashboardPage;
+
+    @Test
+    public void testJenkinsfileCreate() {
+        BitbucketClient client = BitbucketClient.builder()
+            .endPoint("http://127.0.0.1:7990/bitbucket")
+            .credentials("admin:admin").build();
+
+        String project = "PROJECT_1";
+        String repo = "testJenkinsfileCreate";
+        client.api().repositoryApi().delete(project, repo);
+        client.api().repositoryApi().create(project, CreateRepository.create(repo, true));
+
+        dashboardPage.open();
+        wait.until(ExpectedConditions.visibilityOf(dashboardPage.newPipelineButton))
+            .click();;
+        LOGGER.info("Clicked on new pipeline button");
+
+        wait.until(By.xpath("//span[text()='Bitbucket Server']")).click();
+        LOGGER.info("Selected bitbucket server");
+
+        wait.until(By.xpath("//span[text()='Bitbucket Server']")).click();
+        LOGGER.info("Selected bitbucket server");
+
+        wait.until(By.cssSelector(".button-add-server")).click();
+        LOGGER.info("Clicked addserver");
+
+        wait.until(By.xpath("//input[@placeholder='My Bitbucket Server']")).sendKeys("bitbucketserver");
+        wait.until(By.xpath("//input[@placeholder='https://mybitbucket.example.com']")).sendKeys("http://127.0.0.1:7990/bitbucket");
+        wait.until(By.cssSelector(".button-create-server")).click();
+        wait.until(By.cssSelector(".button-next-step")).click();
+        LOGGER.info("Entered server details");
+
+        wait.until(By.xpath("(//input[ @class='TextInput-control'])[1]")).sendKeys("admin");
```

Comment:
Similarly here, any better xpath? Or just css? `.credentials-new input[type=text]` & `.credentials-new input[type=password]`?

Reply:
Nice to have `find(".credentials-new input[type=text]").setText("admin")`